File: packages/exosuit-core/src/models/Idea.ts
import { z } from "zod";

export const IdeaStatusSchema = z.enum([
  "new",
  "triaged",
  "accepted",
  "rejected",
  "deferred",
  "implemented",
  "archived",
]);

export type IdeaStatus = z.infer<typeof IdeaStatusSchema>;

export const IdeaSchema = z.object({
  id: z.string(),
  title: z.string(),
  description: z.string(),
  status: IdeaStatusSchema.default("new"),
  created_at: z.string().datetime(),
  source: z.enum(["user", "agent", "chat-context"]).default("user"),
  tags: z.array(z.string()).default([]),
  related_tasks: z.array(z.string()).default([]),
});

export type Idea = z.infer<typeof IdeaSchema>;

export const IdeasFileSchema = z.object({
  ideas: z.array(IdeaSchema).default([]),
});

export type IdeasFile = z.infer<typeof IdeasFileSchema>;

File: tools/exo/src/idea.rs
use crate::context::Meta;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Idea {
    pub id: String,
    pub title: String,
    pub description: String,
    pub status: String,
    pub created_at: String,
    pub source: String,
    pub tags: Vec<String>,
    pub related_tasks: Vec<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct IdeasFile {
    /// Schema version metadata.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub meta: Option<Meta>,

    pub ideas: Vec<Idea>,
}
